spring-cloud-data-module-deployers/spring-cloud-data-module-deployer-yarn/src/main/java/org/springframework/cloud/data/module/deployer/yarn/DefaultYarnCloudAppService.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.data.module.deployer.yarn;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.StringUtils;
+import org.springframework.yarn.boot.app.YarnContainerClusterApplication;
+import org.springframework.yarn.boot.app.YarnInfoApplication;
+import org.springframework.yarn.boot.app.YarnPushApplication;
+import org.springframework.yarn.boot.app.YarnSubmitApplication;
+
+/**
+ * Default implementation of {@link YarnCloudAppService} which talks to
+ * rest api's exposed by specific yarn controlling container clusters.
+ *
+ * @author Janne Valkealahti
+ */
+public class DefaultYarnCloudAppService implements YarnCloudAppService {
+
+	private static final Logger logger = LoggerFactory.getLogger(DefaultYarnCloudAppService.class);
+	private static final String PREFIX_YIA = "spring.yarn.internal.YarnInfoApplication.";
+	private static final String PREFIX_CCA = "spring.yarn.internal.ContainerClusterApplication.";
+
+	@Override
+	public Collection<CloudAppInfo> getApplications() {
+		ArrayList<CloudAppInfo> infos = new ArrayList<CloudAppInfo>();
+
+		YarnInfoApplication app = new YarnInfoApplication();
+		Properties appProperties = new Properties();
+		appProperties.setProperty(PREFIX_YIA + "operation", "PUSHED");
+		app.appProperties(appProperties);
+		String info = app.run();
+		if (logger.isDebugEnabled()) {
```

Comment:
With SLF4J you can avoid the `if` by doing:

```java
logger.debug("Full response for SUBMITTED: {}", info);
```
In the case where you want to avoid a `toString` invocation, SLF4J will evaluate the log level before constructing the string.

See section **Better yet, use parameterized messages** here:

http://www.slf4j.org/faq.html#logging_performance